processor/src/main/java/org/mapstruct/ap/version/VersionInformation.java

```diff
@@ -0,0 +1,32 @@
+/**
+ *  Copyright 2012-2015 Gunnar Morling (http://www.gunnarmorling.de/)
+ *  and/or other contributors as indicated by the @authors tag. See the
+ *  copyright.txt file in the distribution for a full listing of all
+ *  contributors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package org.mapstruct.ap.version;
+
+/**
+ * Provides information about the processor version and the processor context implementation version
+ *
+ * @author Andreas Gudian
+ */
+public interface VersionInformation {
+    String getRuntimeVersion();
+    String getRuntimeVendor();
+    String getMapStructVersion();
+    String getProcessingEnvironmentImplementationName();
```

Comment:
My intention was to avoid adding any knowledge about specific implementation class names, but of course I can add that so the most common scenarios resolve to something short and snappy. Might break with future JDK / Eclipse JDT releases or with other implementations, though, but I would try to fall-back to the "full" format above.

If that is ok for you, then I'll add it. Just let me know :wink: